### PR TITLE
AWS COGS data pull handler

### PR DESCRIPTION
**Pull request by sudo-buddy**
Please ensure your pull request adheres to the following guidelines:
- [ ] https://jira.corp.adobe.com/browse/SITES-18799
- [ ] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues

Thanks for contributing!

### CODE REVIEW

**Comment by sudo-buddy** [comment]
npm run test is running successfully in local environment but is failing in circleci.

**Comment by sudo-buddy** [comment]
image

File: src/cogs/handler.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  try {
+    const client = new CostExplorerClient({ region });
+    const input = buildAWSInput(startDate, endDate);
+    const command = new GetCostAndUsageCommand(input);
+    const response = await client.send(command);
+    const usageCost = processAWSResponse(response);
+    if (Object.keys(usageCost).length === 0) {
+      return notFound('No Cost Usage found');
+    }
+    Object.keys(usageCost).forEach(async (monthYear) => {
```
Comment (ekremney) [inline src/cogs/handler.js:126]: @sudo-buddy, how does this work when the `startDate` & `endDate` variables span through multiple months partially? For example, `startDate=2024-01-15` and `endDate=2024-02-06`?

**Review by ekremney — COMMENTED** [review]
Implementation-wise LGTM. Just have one question regarding the functionality

File: src/cogs/handler.js (hunk 1 above)
Comment (sudo-buddy) [reply src/cogs/handler.js:126]: @ekremney, the requirement is to get the result by month wise so my assumption was alway get the result for previous whole month.

**Review by ekremney — APPROVED** [review]

**Comment by solaris007** [comment]
closing for now

**Pull request closed without merging**